src/ssl.c

```diff
@@ -2383,6 +2386,11 @@ int wolfSSL_Rehandshake(WOLFSSL* ssl)
         }
 #endif
 
+        if (!resume) {
+            XMEMSET(ssl->session.sessionID, 0, sizeof(ssl->session.sessionID));
```

Comment:
Has this been tested extensively with interop?  The previous full handshake did not require this removal of session.  Why is it required now?

Reply:
I did some more testing and this change wasn't necessary. I reverted some of this change. SecureResume is now a wrapper around Rehandshake, getting and setting the session info.